docs/0.1/grid_product_catalog_family_specification.md

```diff
@@ -0,0 +1,496 @@
+Products.
+
+### Catalog
+
+Catalogs are the primary entities stored in state for the Grid Product Catalog
+smart contract. It has the following fields:
+
+* catalog_id: The unique identifier for a catalog.
+* owner: The ID of the Pike organization that created the catalog.
+* name: A human readable name for the catalog.
+* properties: A list of property values (as defined in Grid Schema) which define
+  additional data about the catalog. These property values are not backed by
+  any schema or property definitions.
+
+```protobuf
+message Catalog {
+    string catalog_id = 1;
+    string owner = 2;
+    string name = 3;
+    repeated PropertyValue properties = 4;
+}
+```
+
+
+### Catalog Product
+
+A catalog product functions as a reference to a product, as defined by the Grid
+Product smart contract. It also defines additional catalog-specific properties
+for the referenced product.
+
+Catalog products re-use the `Product` protobuf message from the Product smart
+contract. It has the following fields:
+
+* product_id: The unique ID for a product. For GS1 products, this is the GTIN.
+  For catalog products, this ID should match the product_id of the referenced
+  product.
+* product_namespace: The namespace of the product for the purposes of checking
+  the properties against the relevant schema. This should match that of the
+  referenced product.
+* owner: The ID of the Pike organization that created the catalog product.
+* properties: A list of additional catalog product attributes, on top of the
+  attributes defined by the product.
+
+```protobuf
+message Product {
+  enum ProductNamespace {
+      UNSET_TYPE = 0;
+      GS1 = 1;
+  }
+
+  string product_id = 1;
+  ProductNamespace product_namespace = 2;
+  string owner = 3;
+  repeated PropertyValue properties = 4;
+}
+```
+
+#### Catalog Product Schema
+
+The list of properties defined by a catalog product must conform to a catalog
+product schema. The Grid Product Catalog smart contract makes several assertions
+about this schema:
+
+* The schema must define a required property with the name `catalog_id`. This
+  property must be a string.
+* The schema must define a required property with the name `status`. This
+  property must be an enum with the options `ACTIVE`, `INACTIVE`, and
+  `DISCONTINUED`.
+* The schema must be named "Catalog Product"
+
+The schema is free to list any other property definitions required for catalog
+products in this Grid network.
+
+The Grid Product Catalog depends on the existence of this schema. Therefore, all
+Grid Product Catalog transactions will be invalid under the following
+circumstances:
+* There is no schema in state with the name "Catalog Product"
+* The schema with the name "Catalog Product" does not have the `catalog_id` or
+  `status` property definitions.
+
+An example "Product Catalog" schema definition is shown below:
+
+```yaml
+- name: "Catalog Product"
+  description: "Schema defining a catalog product"
+  owner: "ACME Corporation"
```

Comment:
This is the ID of the owner, this example may confuse the audience given it feels more like "owner name"